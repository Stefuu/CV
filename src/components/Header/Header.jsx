import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import Flex from '../../components/elements/Flex'
import Box from '../../components/elements/Box'
import Me from '../../assets/me.jpeg'

const Image = styled.img`
    height: 100%;
    border-radius: 50%;
`

const HeaderItem = (props) => (
  <Box
    mt={10}
    style={{ whiteSpace: 'nowrap' }}
    color='#2B547E'
    fontSize={[16, 16, 20]}>
    <span style={{ fontWeight: 'bold' }}>{props.name}: </span>
    <span>{props.text}</span>
  </Box>
)

HeaderItem.propTypes = {
  name: PropTypes.string,
  text: PropTypes.string
}

const Header = (props) => (
  <Flex
    width='100%'
    flexDirection={['column', 'row', 'row']}
    borderBottom='1px solid #e6e6e6'
    pb={40}
    alignItems='center'>
    <Box
      p={15}
      pr={25}
      height={130}>
      <Image src={Me} alt='StÃ©fano' />
    </Box>
    <Box>
      <Box
        mb={15}
        fontSize={[28]}>
        <strong>{props.name}</strong>
        <Box
          fontSize={[23]}>
          <strong>{props.description}</strong>
        </Box>
      </Box>
      <Flex
        flexWrap='wrap'
        flexDirection={['column', 'row', 'row']}>
        {
          props.items.map((item, index) => (
            <Box
              width='50%'
              key={index}>
              <HeaderItem
                name={item.name}
                text={item.text}
              />
            </Box>
          ))
        }
      </Flex>
    </Box>
  </Flex>
)

Header.propTypes = {
  name: PropTypes.string,
  description: PropTypes.string,
  items: PropTypes.array
}

export default Header
