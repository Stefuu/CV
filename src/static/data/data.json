{
	"header": {
		"name": "Stéfano Damiano",
		"description": "Desenvolvedor Front-End",
		"items": [
			{
				"name": "Phone",
				"text": "11 97027-9708"
			},
			{
				"name": "Nasc",
				"text": "08/08/1989"
			},
			{
				"name": "Cidade",
				"text": "São Paulo - SP"
            },
            {
				"name": "Email",
				"text": "mineiro.guaxupe@gmail.com"
			}
		]
    },
    "xperience": {
        "items": [
            {
                "company": "- TOVS (idexo)",
                "duration": "De: Jan/2018 até Hoje",
                "position": "Cargo: Desenvolvedor Front-End Sênior",
                "description": "Descrição: Criação e manutenção de um e-commerce capaz de se adequar a regras de negócio complexas de produtos da empresa",
                "highlights": [                    
                    "- Ambiente de inovação com intenção de renovar a cultura da velha TOTVS. Trabalhamos lado a lado com start-ups incubadas pela TOTVS, o que permite agregar muito sobre tecnologias e soluções muito quentes",
                    "- Tivemos a oportunidade de começar tudo do zero, portanto escolhemos a stack, que é: React com styled-components, styled-system, fazendo uso de context API, Graphql com Relay modern, Node (express) e MQTT (para mensagem entre micro-serviços)",
                    "- Criação de uma arquitetura de execução de regras de negócio no front-end que não são fixas no mesmo. Usamos uma estrutura declarativa com https://github.com/jwadhams/json-logic-js que permite que as regras sejam geradas pelo back-end e apénas consumidas e executadas pelo front (evitando um oceano de condições no front)."
                ]
            },
            {
                "company": "- Universo Online (UOL)",
                "duration": "De: Jul/2012 até Dez/2017",
                "position": "Cargo: Desenvolvedor Front-End Pleno",
                "description": "Descrição: Criação e manutenção de páginas do portal (layout, integrações com back-end e sistemas externos), criação e manutenção de sistemas internos (principalmente em node), criação de serviços",
                "highlights": [                    
                    "- Passei por diversas areas de desenvolvimento da empresa, ganhando bom conhecimento das dificuldades e desafios de colocar e manter um portal com milhões de acessos em produção.",
                    "- Passei pelas dificuldades de se criar layouts responsivos cross-browser que precisam rodar em navegadores antigos.",
                    "- Desenvolvi integrações com ferramentas experimentais do google e facebook. Como o UOL tinha parceria com essas empresas, e era um bom laboratório pela quantidade de acessos, eles liberavam algumas ferramentas novas antes do lançamento. Ferramentas como AMP, Instant Articles e Facebook Messenger BOTS.",
                    "- Aprendi muito sobre arquitetura de back-end trabalhando na criação do micro-serviço do Facebook Messenger BOT, pelo fato de ser necessário fazer envio de milhões de mensagens (quando aparecia alguma notícia bombástica), aprendi muito sobre como escalar uma aplicação node, kubernetes, sobre uso de filas (rabbitmq principalmente) e banco de dados não relacional (mongodb)"
                ]
            },
            {
                "company": "- KAITS - Sistema educacional",
                "duration": "De: Fev/2011 até Jan/2012",
                "position": "Cargo: Estagiário",
                "description": "Descrição: Atuação com javascript vanilla na construção e manutenção de um sistema web para escolas de idioma",
                "highlights": [                    
                    "-Uso de javascript sem nenhum framework foi uma experiência interessante, principalmente por estar começando a usar js na época"                      
                ]
            }
        ]
    },
    "skills": [
        "Javascript",
        "ES20XX",
        "HTML5",
        "CSS3",
        "React",
        "Graphql",
        "Relay",
        "Unity Test",
        "Jest",
        "Enzyme",
        "Nightwatch",
        "Linux",
        "GIT",
        "Docker",
        "Gulp / Grunt",
        "Node",
        "Mongodb",
        "Rabbitmq",
        "SCRUM"
    ],
    "education": {
        "items": [
            {
                "name": "Universidade Presbiteriana Mackenzie",
                "type": "Bacharel em Ciência da Computação",
                "duration": "2008 - 2012"
            }
        ]
    },
    "languages": {
        "items": [
            {
                "name": "Português",
                "level": "Nativo"
            },
            {
                "name": "Inglês",
                "level": "Intermediário"
            },
            {
                "name": "Espanhol",
                "level": "Básico"
            }
        ]
    }
}
