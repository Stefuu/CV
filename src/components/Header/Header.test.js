import React from 'react'
import { render } from 'react-testing-library'
import Header from './Header'
import data from '../../static/data/data.json'

describe('Header', () => {
  it('Deve renderizar [snapshot]', async () => {
    const { container } = render(<Header {...data.header} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
