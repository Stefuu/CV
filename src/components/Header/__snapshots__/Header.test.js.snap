// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Deve renderizar [snapshot] 1`] = `
<div
  class="sc-bdVaJa eXuTnH"
>
  <div
    class="sc-bdVaJa dnIFYn"
  >
    <img
      alt="Stéfano"
      class="sc-bwzfXH dmnngk"
      src="me.jpeg"
    />
  </div>
  <div
    class="sc-bdVaJa hvxHyA"
  >
    <div
      class="sc-bdVaJa heyuGm"
    >
      <strong>
        Stéfano Damiano
      </strong>
      <div
        class="sc-bdVaJa cLkge"
      >
        <strong>
          Desenvolvedor Front-End
        </strong>
      </div>
    </div>
    <div
      class="sc-bdVaJa eVLNKq"
    >
      <div
        class="sc-bdVaJa ccJYFI"
      >
        <div
          class="sc-bdVaJa irusOz"
          style="white-space: nowrap;"
        >
          <span
            style="font-weight: bold;"
          >
            Phone
            : 
          </span>
          <span>
            11 97027-9708
          </span>
        </div>
      </div>
      <div
        class="sc-bdVaJa ccJYFI"
      >
        <div
          class="sc-bdVaJa irusOz"
          style="white-space: nowrap;"
        >
          <span
            style="font-weight: bold;"
          >
            Nasc
            : 
          </span>
          <span>
            08/08/1989
          </span>
        </div>
      </div>
      <div
        class="sc-bdVaJa ccJYFI"
      >
        <div
          class="sc-bdVaJa irusOz"
          style="white-space: nowrap;"
        >
          <span
            style="font-weight: bold;"
          >
            Cidade
            : 
          </span>
          <span>
            São Paulo - SP
          </span>
        </div>
      </div>
      <div
        class="sc-bdVaJa ccJYFI"
      >
        <div
          class="sc-bdVaJa irusOz"
          style="white-space: nowrap;"
        >
          <span
            style="font-weight: bold;"
          >
            Email
            : 
          </span>
          <span>
            mineiro.guaxupe@gmail.com
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
